src/main/java/kg/jarkyn/cobspecserver/ClientSocket.java

```diff
@@ -0,0 +1,28 @@
+package kg.jarkyn.cobspecserver;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+
+public class ClientSocket implements Client {
```

Comment:
I'm finding the name `Client` a little confusing here. A client will be making requests to your HTTP server through their browser, so the fact that the server has an _internal_ client concept I find hard to understand. This class essentially wraps the socket which is fine and does provide a nice layer of abstraction from the rest of the application.